src/plugins/header/snippets/snippets.ts

```diff
@@ -70,8 +70,8 @@ export class UIPSnippets extends UIPPlugin {
     this.render();
     this.bindEvents();
 
-    if (this.model) {
-      this.model.applySnippet(this.model.snippets[this.currentIndex], this);
+    if (this.model && this.model.snippets.length) {
```

Comment:
Again, as in the comment mentioned above, this operation set doesn't look like an area of responsibility for the snippets renderer component.